### PR TITLE
MINOR: Fix potential integer overflow and String.format issue

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
Saw these issues while testing @cmccabe's FindBugs PR. Review by @hachikuji.

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ public void deallocate(ByteBuffer buffer) {
     public long availableMemory() {
         lock.lock();
         try {
-            return this.availableMemory + this.free.size() * this.poolableSize;
+            return this.availableMemory + this.free.size() * (long) this.poolableSize;
```
Comment (apurvam) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java]: So I am curious. The types in this expression are `long + int + int`. Was the result being downcast to an `int`. Will the single cast to `long` fix that, or do both `int`'s need to be cast to `long`?

Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java]: I think you missed that the second expression involves multiplication, not addition. So we need one of the values in the multiplication to be a long.

Comment (apurvam) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java]: Ah. Right. I did miss that!

**Review by apurvam — APPROVED** [review]
LGTM!

**Review by hachikuji — APPROVED** [review]
LGTM

**Pull request closed without merging**